### PR TITLE
Bug: use segment client in Heap directly instead of relying on native modules

### PR DESCRIPTION
**Pull request by storm2513**
## Description

Segment v2 no longer relies on native module for tracking, so previous implementation stopped working

## Changes
- added initialization of segment client in Heap
- added using it for tracking
- added exporting segment client, so we can reuse it in our app
- changed logic to send tracking events only for production build as otherwise we would have wrong data (if someone impersonates staging user with beta app)

### CODE REVIEW

File: js/Heap.js [hunk 1]
```diff
@@ -22,14 +20,25 @@ import { checkDisplayNamePlugin } from './util/checkDisplayNames';
...
+
+const client = createClient({
+  writeKey: Config.SEGMENT_API_KEY,
+  debug: true,
+  trackAppLifecycleEvents: true,
+  flushInterval: 60000, // 60 seconds
+})
+
+// track only for production build as we can change environments in other builds
+const shouldTrack = Config.ENV === 'production'
```
Comment (Wittiest) [inline js/Heap.js]: 💬 Hmmm. I thought it is still useful to track events from admins who are viewing member data. Do you remember the original reason we stopped tracking in non-prod environments?

Comment (storm2513) [reply js/Heap.js]: @Wittiest 

In our app we have this condition:

```
  shouldEnableSegment: () => {
    // only beta and production builds use production segment API key
    if (App.build.beta || App.build.production) {
      return Apollo.endpointEnv === 'production'
    }

    // we don't need to enable it for development endpoint as we have different data locally
    return Apollo.endpointEnv === 'staging'
  },
```

The issue was that we've used staging endpoint with Beta app, so user's data has been updated with data from staging

Comment (storm2513) [reply js/Heap.js]: @Wittiest 
I can add saving  `Apollo.endpointEnv` to AsyncStorage and getting it here, so we have the same logic if it's necessary, wdyt?

Comment (Wittiest) [reply js/Heap.js]: @storm2513 Is there any way to pass a config from the mobile repo into heap?

Comment (Wittiest) [reply js/Heap.js]: As in during initialization of heap or something like this?

Comment (storm2513) [reply js/Heap.js]: @Wittiest updated

**Review by ArtyomTkachGit — APPROVED** [review]

**Review by Wittiest — APPROVED** [review]

**Pull request merged**